Automattic/newspack-theme
# Add a child theme to replace Style 4

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a child theme for Style 4, named Newspack Sacha (after [Sacha Pfeiffer](https://en.wikipedia.org/wiki/Sacha_Pfeiffer); credit to @philipjohn for the name suggestion! 

It also updates the main Newspack theme to use the Sass files from the child theme to build the CSS for style 4. 

There is still some duplicate code: `newspack-theme/inc/color-patterns.php` and `newspack-theme/inc/typography.php` contain a lot of the same code as `newspack-sacha/inc/child-color-patterns.php` and `newspack-sacha/inc/child-typography.php`; this will likely need to stay this way until the style pack code is fully removed from the main theme.

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`; confirm that there are no errors.
2. Navigate to WP Admin > Themes, and confirm the presence of Newspack Sacha. 
3. Try switching to the theme and confirm it loads right; looks like style 4 (note: you will lose a handful of customizer settings on theme switch).
4. Confirm the editor is loading styles correctly.
5. Try setting custom fonts and colours, and confirm they're loading on the front-end and in the editor.
6. Confirm that there is no "Style Packs" option in the Customizer.
6. Switch back to Newspack theme.
7. Confirm that the "Style Packs" option is back in the Customizer.
8. Switch to "Style 4"; confirm that it still works. 
9. Run `npm run release:archive`; confirm that the /releases directory contains both `newspack-theme.zip` and `newspack-sacha.zip`.
10. As an extra precaution, switch back to `master`. Move your generated zip files out of the releases directory, and re-run `npm run release:archive`, and run a file compare of the Newspack theme, to confirm that the generated CSS for Style 4 is the same. 

### Other information:
